src/marching_cubes.jl

```diff
@@ -51,21 +53,21 @@ function marching_cubes(sdf::SignedDistanceField{3,ST,FT},
         # Cube is entirely in/out of the surface
         (cubeindex == 0x00 || cubeindex == 0xff) && continue
 
-        points = (Point{3,Float64}(xi-1,yi-1,zi-1) .* s .+ orig,
-                  Point{3,Float64}(xi,yi-1,zi-1) .* s .+ orig,
-                  Point{3,Float64}(xi,yi,zi-1) .* s .+ orig,
-                  Point{3,Float64}(xi-1,yi,zi-1) .* s .+ orig,
-                  Point{3,Float64}(xi-1,yi-1,zi) .* s .+ orig,
-                  Point{3,Float64}(xi,yi-1,zi) .* s .+ orig,
-                  Point{3,Float64}(xi,yi,zi) .* s .+ orig,
-                  Point{3,Float64}(xi-1,yi,zi) .* s .+ orig)
+        points = (VertType(xi-1,yi-1,zi-1) .* s .+ orig,
+                  VertType(xi,yi-1,zi-1) .* s .+ orig,
+                  VertType(xi,yi,zi-1) .* s .+ orig,
+                  VertType(xi-1,yi,zi-1) .* s .+ orig,
+                  VertType(xi-1,yi-1,zi) .* s .+ orig,
+                  VertType(xi,yi-1,zi) .* s .+ orig,
+                  VertType(xi,yi,zi) .* s .+ orig,
+                  VertType(xi-1,yi,zi) .* s .+ orig)
 
         # Find the vertices where the surface intersects the cube
         find_vertices_interp!(vertlist, points, iso_vals, cubeindex, iso, eps)
 
         # Create the triangle
-        typeof(reduceverts) == Val{true} && _mc_unique_triangles!(vts, fcs, vertlist, cubeindex)
-        typeof(reduceverts) == Val{false} && _mc_create_triangles!(vts, fcs, vertlist, cubeindex)
+        typeof(reduceverts) == Val{true} && _mc_unique_triangles!(vts, fcs, vertlist, cubeindex, FaceType)
```

Comment:
There's probably no benefit to having `reduceverts` being a `Val` type rather than just a boolean.